modules/core/test/v2/unit/recovery/utxo.ts

```diff
@@ -0,0 +1,587 @@
+/**
+ * @prettier
+ */
+import 'should';
+import * as nock from 'nock';
+import * as sinon from 'sinon';
+
+import { TestBitGo } from '../../../lib/test_bitgo';
+import * as config from '../../../../src/config';
+import { Bch, Bsv, Btc } from '../../../../src/v2/coins';
+import {
+  addressInfos,
+  addressUnspents,
+  btcKrsRecoveryDecodedTx,
+  btcNonKrsRecoveryDecodedTx,
+  emptyAddressInfo,
+} from '../../fixtures/coins/recovery';
+import { Bcha } from '../../../../src/v2/coins/bcha';
+import { FixtureDir } from '../../../lib/fixtures';
+nock.disableNetConnect();
+
+const recoveryNocks = require('../../lib/recovery-nocks');
+
+const fixtures = new FixtureDir(`${__dirname}/fixtures`);
+
+describe('UTXO Recovery', function () {
+  let bitgo;
+
+  before(function () {
+    bitgo = new TestBitGo({ env: 'test' });
+    bitgo.initializeTestVars();
+
+    // pretend that Keyternal accepts recoveries for all coins
+    config.krsProviders.keyternal.supportedCoins = ['btc', 'bch', 'bsv', 'bcha', 'ltc', 'zec', 'dash'];
+    (config.krsProviders.keyternal.feeAddresses as any) = {
+      tbtc: '2Mujz9eicmgpPcdScRJTywVK3EQNHDJG3yN',
+      tbch: '2Mujz9eicmgpPcdScRJTywVK3EQNHDJG3yN',
+      tbsv: '2Mujz9eicmgpPcdScRJTywVK3EQNHDJG3yN',
+      tbcha: '2Mujz9eicmgpPcdScRJTywVK3EQNHDJG3yN',
+      tltc: 'QffXMViM8DYgPRf1Hoczjw7BS5CVdSWaBL',
+      tzec: 't2ATLAhBP1uTuyiWs5DY5CPH1VuYkGUindt',
+      tdash: '8euHug4dbmPy3CLawwWdeTjGLqPYEGz3Kt',
+    };
+  });
+
+  describe('Recover Bitcoin', function () {
+    let sandbox: sinon.SinonSandbox;
+
+    beforeEach(() => {
+      sandbox = sinon.createSandbox();
+      recoveryNocks.nockbitcoinFees(600, 600, 100);
+      const callBack1 = sandbox.stub(Btc.prototype, 'getAddressInfoFromExplorer');
+      callBack1.resolves(emptyAddressInfo);
+      callBack1
+        .withArgs('2MzLAGkQVaDiW2Dbm22ETf4ePyLUcDroqdw')
+        .resolves(addressInfos['2MzLAGkQVaDiW2Dbm22ETf4ePyLUcDroqdw']);
+      const callBack2 = sandbox.stub(Btc.prototype, 'getUnspentInfoFromExplorer');
+      callBack2
+        .withArgs('2MzLAGkQVaDiW2Dbm22ETf4ePyLUcDroqdw')
+        .resolves([addressUnspents['2MzLAGkQVaDiW2Dbm22ETf4ePyLUcDroqdw']]);
+      callBack2.resolves([]);
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    it('should generate BTC recovery tx', async function () {
+      sandbox.stub(Btc.prototype, 'verifyRecoveryTransaction').resolves(btcNonKrsRecoveryDecodedTx.transaction);
+      const basecoin = bitgo.coin('tbtc');
+      const recovery = await basecoin.recover({
+        userKey:
+          '{"iv":"fTcRIg7nlCf9fPSR4ID8XQ==","v":1,"iter":10000,"ks":256,"ts":64,"mode":"ccm","adata":"","cipher":"aes","salt":"pkIS5jVDi0Y=","ct":"SJQgP+ZzfOMf2fWxyQ2jpoWYioq6Tqfcw1xiKS1WpWAxLvXfH059sZvPrrYMdijJEbqA8EEaYXWmdgYSkMXdwckRMyvM3uWl9H8iKw1ZJmHyy2eDSy5r/pCtWICkcO3oi2I492I/3Op2YLfIX6XqKWs2mztu/OY="}',
```

Comment:
This `userKey` gets used several times, maybe declare it once as a variable and reuse it?

Looks like it's the same case for `backupKey`, `bitgoKey`, and `recoveryDestination`.

Reply:
yup there is a bunch more we could and should do (loop over coins actually)

I didn't want to go down that rabbithole all the way